### PR TITLE
Options: add a DeathLinkMixin dataclass to easily standardize death_link

### PR DESCRIPTION
**Pull request by alwaysintreble**
## What is this fixing or adding?
title

## How was this tested?
where else would i put it. tested to make sure it assigned correctly.

### CODE REVIEW

**Comment by Jarno458** [comment]
As Suggested by: https://discord.com/channels/731205301247803413/1130599434368057563/1166316118579884065

**Comment by beauxq** [comment]
The name is a little confusing, because it looks like an Option, but it's a collection of Option classes that has one Option in it.

**Comment by alwaysintreble** [comment]
open to a better name, just haven't come up with one

**Review by Exempt-Medic — APPROVED** [review]
LGTM

**Comment by ThePhar** [comment]
I'm not sure I see the ultimate utility of this? Is it just to "enforce" folks use the naming scheme `death_link`?

Most worlds aren't already using this and this just adds expectation to refactor. Plus they'd have to switch away later if they have more complicated Death Link: (e.g. Rogue Legacy, Hollow Knight, Super Metroid)

I think it's fine as it is right now personally.

File: worlds/messenger/options.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ class PlannedShopPrices(OptionDict):
 
 
 @dataclass
-class MessengerOptions(PerGameCommonOptions):
+class MessengerOptions(PerGameCommonOptions, DeathLinkMixin):
```
Comment (black-sliver) [inline worlds/messenger/options.py:136]: I'm pretty sure the mixin should be to the left of the base class.

**Review by black-sliver — COMMENTED** [review]
> I'm not sure I see the ultimate utility of this?

I think Option mixins might be a good idea in general. Currently we have to use getattr for "start_inventory_from_pool" because there is no common type we can "cast" to (i.e. `isinstance(world.options, StartInventoryFromPoolMixin)`.

File: worlds/messenger/options.py (hunk 1 above)
Comment (alwaysintreble) [inline worlds/messenger/options.py:136]:
```suggestion
class MessengerOptions(DeathLinkMixin, PerGameCommonOptions):
```

Comment (black-sliver) [reply worlds/messenger/options.py]: Did you check python docs, or did you blindly believe me? :D

Comment (alwaysintreble) [reply worlds/messenger/options.py]: in our specific case here i don't think it really matters; the order matters for the actual inheritance so there isn't really a "standard" for how it should look, but typically mixins are put before the base class from what i'm able to find. couldn't find any official python docs on mixin specifically though

Comment (black-sliver) [reply worlds/messenger/options.py]: I think left overwrites right, so the more specialized it gets, the farther to the left it should be.

The python docs should mention the inheritance behaviour as MRO, if you have too much free time on your hands :D

**Review by black-sliver — APPROVED** [review]
I'll give other people some time to chime in before hitting merge.

**Pull request merged**